### PR TITLE
Issue #483: Added current configuration to output path

### PR DESCRIPTION
**Pull request by silabs-hfegran**
Signed-off-by: Henrik Fegran <Henrik.Fegran@silabs.com>

### CODE REVIEW

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
Hi @silabs-hfegran, this all looks good.  A comment and a question:
* I've updated #483 with a request for an update to a README to explain these changes.
* Question: which simulator have you tested this with?  Its my understanding that SiLabs Contributors have access to xrun and questa.  If needed I can request other Contributors to test the updates to the VCS and Riviera-PRO makefiles.

**Comment by silabs-hfegran** [comment]
> Hi @silabs-hfegran, this all looks good. A comment and a question:
> 
> * I've updated #483 with a request for an update to a README to explain these changes.
> * Question: which simulator have you tested this with?  Its my understanding that SiLabs Contributors have access to xrun and questa.  If needed I can request other Contributors to test the updates to the VCS and Riviera-PRO makefiles.

Hi @MikeOpenHWGroup, 
That is correct, this has been tested with xrun and questa. I'll add some modifications to the readme as well

**Comment by silabs-hfegran** [comment]
Hi @MikeOpenHWGroup , 

I've updated the Readme (hopefully adding the CV_RESULTS parameter in there does not cause issues as well). 

Just to clarify, the cfg-parameter in itself does not cause a functional difference - It has always been there; the only difference is that we move all simulation outputs run with a given configuration into their respective folders to facilitate multiple simultaneous runs with differing configs. This prevents the simulation environment from being overwritten when running additional simulations with a different configuration.

**Review by MikeOpenHWGroup — APPROVED** [review]

File: mk/uvmt/xrun.mk [hunk 1]
```diff
@@ -85,18 +85,18 @@ endif
...
 endif
 
 ################################################################################
 # Waveform (post-process) command line
 ifeq ($(call IS_YES,$(ADV_DEBUG)),YES)
-WAVES_CMD = cd $(XRUN_RESULTS)/$(TEST_NAME) && $(INDAGO) -db ida.db
+WAVES_CMD = cd $(XRUN_RESULTS)/$(CFG)/$(TEST_NAME) && $(INDAGO) -db ida.db
 else
-WAVES_CMD = cd $(XRUN_RESULTS)/$(TEST_NAME) && $(SIMVISION) waves.shm
+WAVES_CMD = cd $(XRUN_RESULTS)/$(CFG)/$(TEST_NAME) && $(SIMVISION) waves.shm
```
Comment (MikeOpenHWGroup) [inline mk/uvmt/xrun.mk:99]: With the run-index added, I _think_ this should now be:
`WAVES_CMD = cd $(XRUN_RESULTS)/$(CFG)/$(TEST_NAME)_$(RUN_INDEX) && $(SIMVISION) waves.shm`

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
Hi @silabs-hfegran, you are going to hate me for this...

The update to the README looks good (thanks for that), but now that #504 has been merged in, there are multiple conflicts with all the simulator-specific Makefiles.   I have left a comment on `xrun.mk` to illustrate.

I believe I understand the resolution and could probably resolve the conflicts myself, but as I am not 100% sure it is best if you resolve and re-push.

File: mk/uvmt/xrun.mk (hunk 1 above)
Comment (silabs-hfegran) [reply mk/uvmt/xrun.mk:99]: That sounds correct - some care needs to be taken to merge these makefile changes as several of them make changes to the same paths; this as a result of keeping the changes functional on their own and in separate pull-requests.  Feel free to ask if there are any further uncertainties.

Comment (MikeOpenHWGroup) [reply mk/uvmt/xrun.mk:99]: Can I ask you to make the changes?   You are more familiar with the original intent.

Comment (silabs-hfegran) [reply mk/uvmt/xrun.mk:99]: I'm on it, might not get all the changes in before monday though.

**Review by MikeOpenHWGroup — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
Terrific!   Thanks @silabs-hfegran.

**Pull request merged**